### DISCUSSION TITLE
Filtering by message headers in Kafbat UI

### DISCUSSION THREAD

**Discussion by mekkanizer** [kafbat/kafka-ui#552] [Q&A] [answered]
I have a problem when trying to apply a simple filter, I tried both `headers.` and `headers['']` syntax.
For example, `headers.commandName == 'CreateMatch'`
Could you please look into this?
![Screenshot_20240918_170600](https://github.com/user-attachments/assets/ff145b60-a901-4c79-b759-b007d30b7b0c)

**Comment by Haarolean** [comment, marked as answer]
Hi, have you tried consulting with the syntax (upon clicking purple question mark)? This should be `record.headers.xx` rather than `headers.xx`.

```
record.headers.size() == 1 && !has(record.headers.k1) && record.headers['k2'] == 'v2'
```

**Comment by mekkanizer** [comment]
@Haarolean Hello, and thank you for the swift reply. I was merely consulting the official FAQ.
https://ui.docs.kafbat.io/faq/faq#smart-filters-syntax

Again, thank you for resolving this, you are right, I should've noticed the question mark :)
